src/main/java/model/input/file/ExcelFile.java

```diff
@@ -72,19 +86,77 @@ private void addRows(Iterator<Row> rowIterator) {
 		}
 	}
 
-	private DataValue toDataValue(Cell cell) {
+	private DataValue toDataValue(Cell cell, ColumnInfo columnInfo) {
 		switch (cell.getCellType()) {
-			case Cell.CELL_TYPE_STRING:
-				return new StringValue(cell.getStringCellValue());
-			case Cell.CELL_TYPE_NUMERIC:
-				double cellValue = cell.getNumericCellValue();
-				return (cellValue % 1 == 0)
-						? new IntValue((int) cellValue) : new FloatValue((float) cellValue);
 			case Cell.CELL_TYPE_BLANK:
-				return new StringValue("");
-			default: throw new UnsupportedOperationException(
-					String.format("Cell type %s not supported", cell.getCellType())
-			);
+				return DataValue.getNullInstance(columnInfo.getType());
+			case Cell.CELL_TYPE_NUMERIC:
+				return parseNumValue(cell, columnInfo);
+			case Cell.CELL_TYPE_STRING:
+				return parseStringValue(cell, columnInfo);
+			default:
+				throw new UnsupportedOperationException(
+						String.format("Cell type %s not supported", cell.getCellType()));
+		}
+	}
+
+	private DataValue parseStringValue(Cell cell, ColumnInfo columnInfo) {
+		if (isTemporalValue(columnInfo.getType())) {
+			return parseTemporalValue(cell.getStringCellValue(), columnInfo);
+		} else {
+			return new StringValue(cell.getStringCellValue());
+		}
+	}
+
+	private DataValue parseNumValue(Cell cell, ColumnInfo columnInfo) {
+		if (isTemporalValue(columnInfo.getType())) {
+			return parseExcelDateCell(cell, columnInfo);
+		}
+		
+		double cellValue = cell.getNumericCellValue();
+		
+		if (columnInfo.getType() == IntValue.class) {
+			return new IntValue((int) cellValue);
+		} else if (columnInfo.getType() == FloatValue.class) {
+			return new FloatValue((float) cellValue);
+		} else {
+			throw new UnsupportedOperationException(
+				String.format("type %s not supported", columnInfo.getType()));
+		}
+	}
+
+	private DataValue parseExcelDateCell(Cell cell, ColumnInfo columnInfo) {
+		if (DateUtil.isCellDateFormatted(cell)) {
+			Date date = cell.getDateCellValue();
+			return parseDateCellValue(date, columnInfo.getType());
+	
+		} else {
+			if (columnInfo.getFormat().equals(EXCEL_DATE)) {
+				Date javaDate = DateUtil.getJavaDate(cell.getNumericCellValue());
+				return parseDateCellValue(javaDate, columnInfo.getType());
+			} else {
+				return parseTemporalValue(
+						String.valueOf(cell.getNumericCellValue()), columnInfo);
+			}
+		}
+	}
+
+	private DataValue parseDateCellValue(Date date, Class<? extends DataValue> type) {
+		if (type == DateTimeValue.class) {
+			Instant instant = Instant.ofEpochMilli(date.getTime());
+			return new DateTimeValue(LocalDateTime.ofInstant(instant, ZoneId.systemDefault()));
+
+		} else if (type == DateValue.class) {
+			Instant instant = Instant.ofEpochMilli(date.getTime());
+			return new DateValue(LocalDateTime.ofInstant(instant, ZoneId.systemDefault()));
+
+		} else if (type == TimeValue.class) {
+			Calendar calendar = Calendar.getInstance();
+			calendar.setTime(date);
```

Comment:
Dan doe je toch LocalDateTime.toLocalTime()? Het is nu echt super vaag dat we overal java.time gebruiken, noodgedwongen de java.util.Date moeten gebruiken omdat we die uit een library krijgen en dan java.util.Calendar gebruiken omdat we niet even de moeite willen nemen om het in java.time te doen.